src/apps/replication/meta_server/meta_state_service_simple.cpp

```diff
@@ -275,33 +283,33 @@ namespace dsn
             binary_writer log_writer;
             int magic = 0xdeadbeef;
             log_writer.write(magic);
-            log_writer.write(static_cast<size_t>(log_writer.total_size()));
-            log_writer.write(log_writer.get_buffer());
-            
+            log_writer.write(temperal_writer.get_buffer());
+
```

Comment:
Because otherwise I have to reserve a slot for 'size' and fill it when I actual know the size of the buffer. I considered this to be ugly when I wrote the code. I'm rewriting this one in the next commit.